src/lobby.rs

```diff
@@ -22,13 +23,20 @@ pub struct Lobby {
     inner: Arc<Mutex<Inner>>,
 }
 
+#[derive(Clone)]
```

Comment:
Should I derive more things? Like Copy? What does that get me?

Reply:
derive what you need basically. `Debug` is a common thing to want for printing debug strings. You cannot derive Copy because HashSets aren't copy.

https://doc.rust-lang.org/book/appendix-03-derivable-traits.html?highlight=copy#clone-and-copy-for-duplicating-values